Kas updates (#129)
bisdn/bisdn-linux: README.md

# BISDN Linux build system

This repository holds files for building the
[BISDN Linux](https://docs.bisdn.de/) network operating system.

Pre-built release images can be downloaded from the
[BISDN Linux images](https://docs.bisdn.de/download_images.html) page.
[Nightly builds](http://repo.bisdn.de/nightly_builds/) are also available.

If you want to build your own BISDN Linux images, you can use the files

> in this repo and the kas automation tool. The only file that you will

Reviewer: ```suggestion
in this repo and the [kas](https://github.com/siemens/kas) automation tool. The only file that you will
```

Author: Done.

need is the kas configuration file `bisdn-linux.yaml`. All other files
in this repo are documentation or for the benefit of our CI/CD and
release management systems.

The remainder of this document provides a quick guide on how to build
a BISDN Linux image.

## Supported switches

A list of supported switches can be found at
[BISDN Linux images](https://docs.bisdn.de/download_images.html).

## Build requirements

### Hardware requirements

> We recommend a build host with 16 CPU cores, 32 GiB RAM and 150 GiB free

Reviewer: ```suggestion
We recommend a build host with 32 CPU cores, 32 GiB RAM and 150 GiB free
```

Reviewer: to match the time and disk space section below

Author: Done.

disk space for building a full image. You may get away with less, but
you risk running into out-of-memory errors, a full disk, or a slow build.

### Software requirements

Our build system uses
[kas](https://kas.readthedocs.io/en/latest/userguide.html) to fetch the
source repositories and to configure the build. While it is possible to

> run `kas` directly on your build host, we recommend using `kas-container`

Reviewer: ```suggestion
run `kas` directly on your build host, we recommend using [kas-container](https://kas.readthedocs.io/en/latest/userguide/kas-container.html)
```

`build` directory tend to grow slowly when building new versions.

## Building a BISDN Linux image

### x86_64 platforms

The default image works on all x86_64 platforms supported by BISDN
Linux. To build the image from the latest sources, clone this repo and
run the following command in its root directory:

```bash
kas-container build bisdn-linux.yaml
```

After the build process has finished, your image will be in
`build/tmp/deploy/images/generic-x86-64/`. A symlink named
`onie-bisdn-full-generic-x86-64.bin` will point to the actual image
file named `onie-bisdn-full-generic-x86-64-<timestamp>.bin`.

### ARM platforms (Edgecore AS4610)

To build an image that works on all ARM platforms supported by BISDN Linux,
run this command in the root of this repository:

```bash
MACHINE=generic-armel-iproc kas-container build bisdn-linux.yaml
```

After the build process finishes, your image will be in
`build/tmp/deploy/images/generic-armel-iproc/`. A symlink named
`onie-bisdn-full-generic-armel-iproc.bin` will point to the actual image
file named `onie-bisdn-full-generic-armel-iproc-<timestamp>.bin`.

### ofdpa-gitlab

> If you want to build ofdpa from source and you have access to the ofdpa-gitlab
> repo, the most convenient way is to use your `.ssh` directory (to avoid
> complaints that the authenticity of the host can't be established) and your
> ssh-agent (for authentication with gitlab.bisdn.de):

Reviewer: ```suggestion
If you want to build ofdpa from source, you need access to the ofdpa-gitlab
repo. The most convenient way to authenticate, while building inside the kas-container,
is to use your own `.ssh` directory and ssh-agent (to avoid
complaints that the authenticity of the host can't be established), by passing the command line
arguments as shown below:
```

Author: I kept some of the original phrasing (the information in parentheses).
